### PR TITLE
fix: setting ui

### PR DESCRIPTION
**Pull request by beyond009**
https://mask.atlassian.net/browse/MF-5722
https://mask.atlassian.net/browse/MF-5703
https://mask.atlassian.net/browse/MF-5705
https://mask.atlassian.net/browse/MF-5661
https://mask.atlassian.net/browse/MF-5728
https://mask.atlassian.net/browse/MF-5658
https://mask.atlassian.net/browse/MF-5704
https://mask.atlassian.net/browse/MF-5731

### CODE REVIEW

File: src/modals/LoginModal.tsx [hunk 1]
```diff
@@ -105,9 +104,7 @@ export const LoginModal = forwardRef<SingletonModalRefCreator<LoginModalProps |
                                                     onClick={async () => {
                                                         if (platform === SocialPlatform.Lens) {
                                                             try {
-                                                                await getWalletClientRequired({
-                                                                    chainId: polygon.id,
-                                                                });
+                                                                await getWalletClientRequired();
```
Comment (guanbinrui) [inline src/modals/LoginModal.tsx:107]: Why?

File: src/components/AccountSetting/FarcasterAccountSetting.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import UserAddIcon from '@/assets/user-add.svg';
+import { PlatformIcon } from '@/components/PlatformIcon.js';
+import { SocialPlatform } from '@/constants/enum.js';
+import { Image } from '@/esm/Image.js';
+import { LoginModalRef, LogoutModalRef } from '@/modals/controls.js';
+import { useFarcasterStateStore } from '@/store/useFarcasterStore.js';
+
+export function FarcasterAccountSetting() {
+    const { disconnect } = useDisconnect();
+    const farcasterAccounts = useFarcasterStateStore.use.profiles();
```
Comment (guanbinrui) [inline src/components/AccountSetting/FarcasterAccountSetting.tsx:16]: We don't use `account` anymore. Please use `profile` instead.

File: src/components/AccountSetting/FarcasterAccountSetting.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                                    className="absolute left-[24px] top-[24px] h-[16px] w-[16px] rounded-[99px] border border-white shadow"
+                                    platform={SocialPlatform.Farcaster}
+                                    size={16}
+                                />
+                            </div>
+                        </div>
+                        <div className="inline-flex h-[39px] shrink grow basis-0 flex-col items-start justify-center">
+                            <div className=" text-[15px] font-medium text-main">{displayName}</div>
+                            <div className=" text-[15px] font-normal text-lightSecond">@{profileId}</div>
+                        </div>
```
Comment (guanbinrui) [inline src/components/AccountSetting/FarcasterAccountSetting.tsx:40]: Use `<ProfileName />`.

File: src/components/AccountSetting/FarcasterAccountSetting.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                        {currentAccount && currentAccount.profileId === profileId ? (
+                            <div
+                                className="h-[8px] w-[8px] rounded-[99px] bg-[#3DC233]"
```
Comment (guanbinrui) [inline src/components/AccountSetting/FarcasterAccountSetting.tsx:43]: Avoid dead-code hex string.

File: src/components/AccountSetting/FarcasterAccountSetting.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                                style={{ filter: 'drop-shadow(0px 4px 10px #3DC233)' }}
+                            />
+                        ) : null}
+                    </div>
+                ))}
+                <button
+                    className="flex w-full items-center gap-[8px]"
+                    onClick={() => {
+                        disconnect();
```
Comment (guanbinrui) [inline src/components/AccountSetting/FarcasterAccountSetting.tsx:52]: Why need to `discconnect` wallet first?

File: src/components/AccountSetting/LensAccountSetting.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+                                    className="absolute left-[24px] top-[24px] h-[16px] w-[16px] rounded-[99px] border border-white shadow"
+                                    platform={SocialPlatform.Lens}
+                                    size={16}
+                                />
+                            </div>
+                        </div>
+                        <div className="inline-flex h-[39px] shrink grow basis-0 flex-col items-start justify-center">
+                            <div className=" text-[15px] font-medium text-main">{displayName}</div>
+                            <div className=" text-[15px] font-normal text-lightSecond">@{profileId}</div>
+                        </div>
```
Comment (guanbinrui) [inline src/components/AccountSetting/LensAccountSetting.tsx:40]: Use `<ProfileAvatar />` and `<ProfileName />`.

File: src/components/AccountSetting.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+                            />
+                        ) : null}
+                    </div>
+                ))}
+                <button
+                    className="flex w-full items-center gap-[8px]"
+                    onClick={() => {
+                        if (platform === SocialPlatform.Lens) disconnect();
+                        clearCurrentProfile();
+                        LoginModalRef.open({});
```
Comment (guanbinrui) [inline src/components/AccountSetting.tsx:41]:
```suggestion
                        LoginModalRef.open();
```

File: src/components/Login/LoginLens.tsx [hunk 8]
```diff
@@ -14,6 +14,7 @@ import WalletIcon from '@/assets/wallet.svg';
 import { AccountCard } from '@/components/Login/AccountCard.js';
 import { EMPTY_LIST } from '@/constants/index.js';
 import { useCustomSnackbar } from '@/hooks/useCustomSnackbar.js';
+import { isSameAddress } from '@/maskbook/packages/web3-shared/base/src/index.js';
```
Comment (guanbinrui) [inline src/components/Login/LoginLens.tsx:17]:
```suggestion
import { isSameAddress } from '@masknet/web3-shared-base';
```

File: src/app/(settings)/settings/display/page.tsx [hunk 9]
```diff
@@ -56,16 +66,18 @@ export default function Display() {
                     setThemeMode(Appearance.dark);
                 }}
             >
-                {mode === Appearance.dark ? (
-                    <div className="relative h-2 w-2 shadow">
+                <div className="flex items-center gap-[20px]">
+                    {mode === Appearance.dark ? (
                         <div
-                            className="absolute left-0 top-0 h-2 w-2 rounded-full bg-[#3DC233]"
+                            className="h-2 w-2 rounded-full bg-[#3DC233]"
```
Comment (guanbinrui) [inline src/app/(settings)/settings/display/page.tsx:72]: Avoid use hex color.

File: src/app/(settings)/settings/display/page.tsx [hunk 10]
```diff
@@ -2,34 +2,42 @@
...
                     setThemeMode(Appearance.default);
                 }}
             >
-                {mode === Appearance.default ? (
-                    <div className="relative h-2 w-2 shadow">
+                <div className="flex items-center gap-[20px]">
+                    {mode === Appearance.default ? (
                         <div
-                            className="absolute left-0 top-0 h-2 w-2 rounded-full bg-[#3DC233]"
+                            className="h-2 w-2 rounded-full bg-[#3DC233]"
```
Comment (guanbinrui) [inline src/app/(settings)/settings/display/page.tsx:32]: Avoid using dead-coded hex color.

File: src/app/(settings)/settings/display/page.tsx [hunk 11]
```diff
@@ -2,34 +2,42 @@
...
+                        ? ' border border-white bg-slate-950 text-white'
+                        : 'border border-neutral-900 bg-white text-slate-950'
+                }`}
                 onClick={() => {
                     setThemeMode(Appearance.default);
                 }}
             >
-                {mode === Appearance.default ? (
-                    <div className="relative h-2 w-2 shadow">
+                <div className="flex items-center gap-[20px]">
```
Comment (guanbinrui) [inline src/app/(settings)/settings/display/page.tsx:29]: This component is repeating three times.

Comment (beyond009) [reply src/app/(settings)/settings/display/page.tsx:29]: They have different styles, maybe it's better to keep this way

**Review by guanbinrui — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**